updateStableIds/src/main/java/org/reactome/release/updateStableIds/UpdateStableIds.java

```diff
@@ -0,0 +1,171 @@
+		GKInstance sliceIE = InstanceEditUtils.createInstanceEdit(dbaSlice, personId, creatorName);
+		GKInstance gkCentralIE = InstanceEditUtils.createInstanceEdit(dbaGkCentral, personId, creatorName);
+
+		// At time of writing (December 2018), test_slice is a non-transactional database. This check has been put in place as a safety net in case that changes.
+		if (dbaSlice.supportsTransactions()) {
+			dbaSlice.startTransaction();
+		}
+		dbaGkCentral.startTransaction();
+
+		//TODO: Perl wrapper will create a 'snapshot' of the previous slice -- once the wrapper is retired this needs to be done
+		
+		// These two Lists were originally used to determine what classes' instances should be updated.
+		// Since all 'accepted' classes turned out to encompass all PhysicalEntity and Event classes, and all
+		// 'excluded' classes were Regulation classes, the filter was not needed anymore.
+//		final List<String> acceptedClassesWithStableIdentifiers = Arrays.asList("Pathway","SimpleEntity","OtherEntity","DefinedSet","Complex","EntityWithAccessionedSequence","GenomeEncodedEntity","Reaction","BlackBoxEvent","CandidateSet","Polymer","Depolymerisation","Polymerisation","Drug","FailedReaction","EntitySet");
+//		List<String> excludedClassesWithStableIdentifiers = Arrays.asList("PositiveGeneExpressionRegulation","PositiveRegulation","NegativeRegulation","Requirement","NegativeGeneExpressionRegulation");
+//		ResultSet classesWithStableIdentifiers = dbaGkCentral.executeQuery("SELECT DISTINCT _class FROM DatabaseObject WHERE StableIdentifier IS NOT NULL", null);		
+
+		logger.info("Fetching all Event and PhysicalEntity instances");
+		// Get all Event and PhysicalEntity instances and combine them into one large List
+		Collection<GKInstance> eventInstances = dbaSlice.fetchInstancesByClass(Event);
+		Collection<GKInstance> physicalEntityInstances = dbaSlice.fetchInstancesByClass(PhysicalEntity);
+		List<GKInstance> sliceInstances = new ArrayList<GKInstance>();
+		sliceInstances.addAll(eventInstances);
+		sliceInstances.addAll(physicalEntityInstances);
+
+		int incrementedCount = 0;
+		int notIncrementedCount = 0;
+		logger.info("Total instances to check: " + sliceInstances.size());
+
+		for (GKInstance sliceInstance : sliceInstances) {
+			logger.info("Checking " + sliceInstance);
+			GKInstance gkCentralInstance = dbaGkCentral.fetchInstance(sliceInstance.getDBID());
+			GKInstance prevSliceInstance = dbaPrevSlice.fetchInstance(sliceInstance.getDBID());
+			// Check if instance is new and that it exists on gkCentral (they could be deleted)
+			if (prevSliceInstance != null && gkCentralInstance != null) {
+				
+				// Compare number of 'Modified' instances between slices
+				Collection<GKInstance> sliceInstanceModified = sliceInstance.getAttributeValuesList(modified);
+				Collection<GKInstance> prevSliceInstanceModified = prevSliceInstance.getAttributeValuesList(modified);
+				if (sliceInstanceModified.size() > prevSliceInstanceModified.size()) {
```

Comment:
Is it _certain_ that `sliceInstanceModified` & `prevSliceInstanceModified` are non-null at this point? If an instance has no `modified`s, will an empty list be returned or null? Is this something that needs to be checked? Or will this code only run for things that have `modified`s? I would have thought that a stableIdentifier that was brand-new in the previous release might _not_ have anything in its `modified` attribute - but maybe it's OK if `getAttributeValuesList` returns an empty list instead of NULL.

Reply:
It returns an empty list, which I think preserves what the code is attempting to do